src/Components/LinkCreator/LinkCreator.jsx

```diff
@@ -1,49 +1,44 @@
-import React, { useState, useEffect } from "react";
+import { useState, useEffect } from "react";
 import { useSelector } from "react-redux";
 
 import { checkUrlAvailable } from "../../utils/api.js";
 import { vocascanServer } from "../../utils/constants.js";
 
-import "./LinkCreator.scss";
-
-const LinkCreator = ({ path, setValid, electronFix = false, children }) => {
+const LinkCreator = ({ path, children, electronFix = false }) => {
   const selfHosted = useSelector((state) => state.login.selfHosted);
   const serverAddress = useSelector((state) => state.login.serverAddress);
   const appLanguage = useSelector((state) => state.setting.language);
 
+  const [isValid, setIsValid] = useState(false);
   const [url, setUrl] = useState("");
 
   const { BASE_URL: baseURL, ENV: env } = window.VOCASCAN_CONFIG;
 
-  //initialize url with the needed url for the servers
+  // initialize url with the needed url for the servers
   useEffect(() => {
-    setUrl(
-      baseURL || selfHosted
-        ? electronFix && env === "electron"
-          ? vocascanServer + `${path}?lang=` + appLanguage
-          : (serverAddress ? serverAddress : baseURL) +
-            `${path}?lang=` +
-            appLanguage
-        : vocascanServer + `${path}?lang=` + appLanguage
-    );
+    if (baseURL || selfHosted) {
+      if (electronFix && env === "electron") {
+        setUrl(vocascanServer + `${path}?lang=` + appLanguage);
+      } else {
+        setUrl((serverAddress || baseURL) + `${path}?lang=` + appLanguage);
+      }
+    } else {
+      setUrl(vocascanServer + `${path}?lang=` + appLanguage);
+    }
   }, [appLanguage, baseURL, electronFix, env, path, selfHosted, serverAddress]);
 
   //check if url is available
```

Comment:
i am very accurate here :joy:  - I hope that does not bother you

Reply:
what should I change there?

Reply:
the space between the `//` and the content of the comment

Reply:
Ah I see. Will fix it later